sources/directconnect/virturl-interface.go

```diff
@@ -0,0 +1,187 @@
+			Tags:            tagsToMap(virtualInterface.Tags),
+		}
+
+		if virtualInterface.ConnectionId != nil {
+			// +overmind:link directconnect-connection
+			item.LinkedItemQueries = append(item.LinkedItemQueries, &sdp.LinkedItemQuery{
+				Query: &sdp.Query{
+					Type:   "directconnect-connection",
+					Method: sdp.QueryMethod_GET,
+					Query:  *virtualInterface.ConnectionId,
+					Scope:  scope,
+				},
+				BlastPropagation: &sdp.BlastPropagation{
+					// We cannot delete a connection if it has virtual interfaces
+					In: true,
+					// We can't affect the connection
+					Out: false,
+				},
+			})
+		}
+
+		if virtualInterface.DirectConnectGatewayId != nil {
+			// +overmind:link directconnect-direct-connect-gateway
+			item.LinkedItemQueries = append(item.LinkedItemQueries, &sdp.LinkedItemQuery{
+				Query: &sdp.Query{
+					Type:   "directconnect-direct-connect-gateway",
+					Method: sdp.QueryMethod_GET,
+					Query:  *virtualInterface.DirectConnectGatewayId,
+					Scope:  "global",
+				},
+				BlastPropagation: &sdp.BlastPropagation{
+					// We cannot delete a direct connect gateway if it has virtual interfaces
+					In: true,
+					// We can't affect the direct connect gateway
+					Out: false,
+				},
+			})
+		}
+
+		if virtualInterface.AmazonAddress != nil {
+			// +overmind:link ip
+			item.LinkedItemQueries = append(item.LinkedItemQueries, &sdp.LinkedItemQuery{
+				Query: &sdp.Query{
+					Type:   "ip",
+					Method: sdp.QueryMethod_GET,
+					Query:  *virtualInterface.AmazonAddress,
+					Scope:  "global",
+				},
+				BlastPropagation: &sdp.BlastPropagation{
+					// IPs are always linked
+					In:  true,
+					Out: true,
+				},
+			})
+		}
+
+		if virtualInterface.CustomerAddress != nil {
+			// +overmind:link ip
+			item.LinkedItemQueries = append(item.LinkedItemQueries, &sdp.LinkedItemQuery{
+				Query: &sdp.Query{
+					Type:   "ip",
+					Method: sdp.QueryMethod_GET,
+					Query:  *virtualInterface.CustomerAddress,
+					Scope:  "global",
+				},
+				BlastPropagation: &sdp.BlastPropagation{
+					// IPs are always linked
+					In:  true,
+					Out: true,
+				},
+			})
+		}
+
+		// Pinpoint a single attachment
+		if virtualInterface.DirectConnectGatewayId != nil && virtualInterface.VirtualInterfaceId != nil {
+			// +overmind:link directconnect-direct-connect-gateway-attachment
+			item.LinkedItemQueries = append(item.LinkedItemQueries, &sdp.LinkedItemQuery{
+				Query: &sdp.Query{
+					Type:   "directconnect-direct-connect-gateway-attachment",
+					Method: sdp.QueryMethod_GET,
+					// returns a single attachment
+					// TODO: implement this format on the directconnect-direct-connect-gateway-attachment source for GET
+					Query: fmt.Sprintf(gatewayIDVirtualInterfaceIDFmt, *virtualInterface.DirectConnectGatewayId, *virtualInterface.VirtualInterfaceId),
+					Scope: scope,
+				},
+				BlastPropagation: &sdp.BlastPropagation{
+					// Changes to the attachment won't affect virtual interface
+					In: false,
```

Comment:
Once again, because the attachment is a kind of "bridging" resource (like an IP often is) I'd argue it's worth having this as `true` so that changes to the direct connect gateway propagate through the attachment to the interface

```suggestion
					In: true,
```